examples/ui-demo/src/state/store.tsx

```diff
@@ -0,0 +1,212 @@
+            setSupportUrl,
+            setTheme,
+            setNftTransferred,
+            nftTransferred,
+            ...config
+          }) => config,
+          skipHydration: true,
+          version: 1,
+        })
+  );
+};
+
+function createInitialState(
+  initialConfig: Config = DEFAULT_CONFIG
+): StateCreator<DemoState> {
+  return (set, get) => ({
+    ...initialConfig,
+    nftTransferred: false,
+    setNftTransferred: (transferred) => {
+      set({ nftTransferred: transferred });
+    },
+    setTheme: (theme) => {
+      set((state) => ({
+        ui: {
+          ...state.ui,
+          theme,
+        },
+      }));
+    },
+    setSupportUrl: (url) => {
+      set(() => ({
+        supportUrl: url,
+      }));
+    },
+    setLogo: (theme, logo) => {
+      set((state) => ({
+        ui: {
+          ...state.ui,
+          [theme]: logo,
+        },
+      }));
+    },
+    setPrimaryColor: (theme, color) => {
+      set((state) => ({
+        ui: {
+          ...state.ui,
+          primaryColor: {
+            ...state.ui.primaryColor,
+            [theme]: color,
+          },
+        },
+      }));
+    },
+    setAuth: (auth) => {
+      set((state) => ({
+        auth: {
+          ...state.auth,
+          ...auth,
+        },
+      }));
+    },
+    setIllustrationStyle: (style: Config["ui"]["illustrationStyle"]) => {
+      set((state) => ({
+        ui: {
+          ...state.ui,
+          illustrationStyle: style,
+        },
+      }));
+    },
+    setBorderRadius: (radius: Config["ui"]["borderRadius"]) => {
+      set((state) => ({
+        ui: {
+          ...state.ui,
+          borderRadius: radius,
+        },
+      }));
+    },
+  });
+}
+
+function deserialize<type>(value: string): type {
+  return JSON.parse(decodeURIComponent(value));
+}
+
+export function cookieToInitialConfig(cookie?: string | null) {
+  if (!cookie) return;
+
+  const state = parseCookie(cookie, STATE_KEY);
+  if (!state) return;
+
+  return deserialize<{ state: Config }>(state).state;
+}
+
+export function generateStylesForRoot(config: Config) {
+  const primaryColor = config.ui.primaryColor[config.ui.theme];
+
+  return [
+    [getColorVariableName("fg-accent-brand"), primaryColor],
+    [getColorVariableName("btn-primary"), primaryColor],
+    [
+      getBorderRadiusBaseVariableName(),
+      getBorderRadiusValue(config.ui.borderRadius),
+    ],
+  ];
+}
+
+export function generateClassesForRoot(config: Config): string[] {
+  return [config.ui.theme];
+}
+
+export function deepEqual(a: any, b: any): boolean {
+  if (typeof a !== typeof b) {
+    return false;
+  }
+
+  if (Array.isArray(a) && Array.isArray(b)) {
+    if (a.length !== b.length) {
+      return false;
+    }
+    for (let i = 0; i < a.length; i++) {
+      if (!deepEqual(a[i], b[i])) {
+        return false;
+      }
+    }
+    return true;
+  } else if (
+    typeof a === "object" &&
+    typeof b === "object" &&
+    a != null &&
+    b != null
```

Comment:
This still has `deepEqual({}, [])` returning true because `typeof {} === typeof [] === "object"` and `Object.keys({}) === Object.keys([]) === []`.

Reply:
the check before this if statement though does the `Array.isArray(a) && Array.isArray(b)` though no? or is there another check that needs to happen up there?

Reply:
oh wait no, I get it. fixing